localbot/AnonModule.cs

```diff
@@ -11,61 +11,102 @@ namespace localbot
 {
     public class AnonModule : ModuleBase<SocketCommandContext>
     {
+        // TODO:
+        // - reduce redundancy of checks when executing commands
+        // 
+        // - fix so not all commands can be executed in any context
+        // 
+        // - admin role that can be set instead of actual admin perms
+        //
+        // - command to set the history length and cooldown time for 
+        // newID and anonUser history respectivly
+
+        // Max ID number to be generated or chosen with >newID
         public const int maxID = 1000;
+        // Length of the history of IDs that each AnonUser records
         public const int historyLength = 5;
 
         private class AnonUser
         {
-            public ulong user { get; set; }
-            private Stack<int> ids;
-            public DateTime lastNewID { get; set; }
+            public ulong user;
+            private List<int> ids;
+            public DateTime lastNewID;
+            public bool blacklisted;
 
+            public bool timeout;
+            public DateTime timeoutEnd;
+
+            // Takes an unsigned long user and returns a AnonUser with 
+            // user set to the user
             public AnonUser(ulong user)
             {
                 this.user = user;
-                ids = new Stack<int>();
+                ids = new List<int>();
             }
 
+            // Returns the current ID number that the AnonUser has set
             public int getID()
             {
-                return ids.Peek();
+                return ids[0];
             }
 
-            public bool AliasAs(int id)
+            // Returns true if this AnonUser is blacklisted
+            public bool IsBlacklisted()
             {
-                bool foundID = false;
-                Stack<int> temp = new Stack<int>();
-                while(ids.Count != 0)
+                return blacklisted;
+            }
+
+            // Returns true if this user is timed out
+            public bool IsTimedout()
+            {
+                if (timeout == false)
```

Comment:
```suggestion
                return timeout ? (timeoutEnd - DateTime.Now) > new TimeSpan(0) : false;
```

You could replace this method with a ternary operator for readability